### DISCUSSION TITLE
Optional catch-all routes - How are they supposed to work with SSG? Is it a SSR-only feature?

### DISCUSSION THREAD

**Discussion by Vadorequest** [vercel/next.js#16907] [Help] [answered]
I wanted to play around with the latest https://nextjs.org/blog/next-9-5#optional-catch-all-routes feature, and I blindly used a SSG page to do so.

I ended up with [`src/pages/[locale]/examples/native-features/example-optional-catch-all-routes/[[...slug]].tsx`](https://github.com/UnlyEd/next-right-now/pull/162/files#diff-e50f7b468a653db6b9a4172ef3cb54afR35-R49):
```tsx
type PageAdditionalServerSideParams = {
  slug: string[];
}

export const getStaticPaths: GetStaticPaths<CommonServerSideParams> = async (): Promise<StaticPathsOutput<PageAdditionalServerSideParams>> => {
  const paths: StaticPath<PageAdditionalServerSideParams>[] = map(supportedLocales, (supportedLocale: I18nLocale): StaticPath<PageAdditionalServerSideParams> => {
    return {
      params: {
        locale: supportedLocale.name,
        slug: [],
      },
    };
  });

  return {
    fallback: false,
    paths,
  };
};
```

I didn't think much before trying it out, but it doesn't work properly and I wonder if that's even compatible with SSG.

If I don't add a `slug` params in `getStaticPaths` then the whole build fails. If I add it, then it'll only match the slugs I've specified. 🤔 

I have to use `getStaticPaths`, because all my pages start with `/[locale]`. 

I thought the point was to match any dynamic path, what did I miss?

**Comment by Vadorequest** [comment]
@Timer If you have any feedback on this, that would be great!

**Comment by lfades** [comment, marked as answer]
@Vadorequest You want to use [`fallback: 'true'`](https://nextjs.org/docs/basic-features/data-fetching#fallback-true) if you want it to handle every path, including those not returned by `getStaticPaths`.

**Reply by Vadorequest** [reply]
Thanks! I'll give it a try ASAP. I hope it's that simple 😅

**Reply by Vadorequest** [reply]
I can confirm this works fine! 

See PR at https://github.com/UnlyEd/next-right-now/pull/162
Demo at https://nrn-v2-mst-aptd-at-lcz-sty-c1.vercel.app/fr/examples/native-features/example-optional-catch-all-routes

**Reply by watch-janick** [reply]
Can you make it work with `next export` and configure the CDN to redirect correctly to the catch-all routes?

I've created a discussion about it here: https://github.com/vercel/next.js/discussions/17711

Thanks! 🙏
